tests/test_poincare_math.py

```diff
@@ -370,3 +370,17 @@ def test_distance2plane(a, c):
     dist = poincare.math.dist2plane(z, a, vr, c=c)
 
     np.testing.assert_allclose(dist, dist1, atol=1e-5, rtol=1e-5)
+
+
+def test_klein_transition(a, c):
+    kl = poincare.math.poincare2klein(a, c=c)
+    b = poincare.math.klein2poincare(kl, c=c)
+    np.testing.assert_allclose(a, b, atol=1e-6, rtol=1e-6)
+
+
+def test_midpoint(a, c):
+    # test the midpoint with zero curvature
+    mids_a = torch.mean(a, dim=-1)
+    mids_b = poincare.math.midpoint(a, c=0, dim=-1)
+    np.testing.assert_allclose(mids_a, mids_b, atol=1e-6, rtol=1e-6)
```

Comment:
@ferrine this test doesn't pass. `poincare.math.midpoint` output is `mids_b = 10 * mids_a`.

Reply:
Consider also adding a test that barycentre of a measure concentrated on two points lies on a geodesic connecting these points

Reply:
So far there are concerns that they indeed lie there

Reply:
I've checked this for poincare ball of dimension 2 ant it was all right, intuition breaks for dim>2 as in test

Reply:
Remark for above. Confirmed for dim>2 and 2 points it is ok once points are not close to the boundary. From more than 2 points running and optimization to find barycenter ($argmin_y \mathbb{E}d(y, x)^2$) gives different result than poincare_mean. Checking loss values for these 2 points there is a confidence that optimization has converged and found the barycenter.